.github/workflows/deploy-image-on-master-merge.yaml

```diff
@@ -0,0 +1,82 @@
+name: Deploy Application images when Merging on Master
+
+on:
+  # run it on push to the default repository branch
+  push:
+    branches: [master]
+
+env:
+  REGISTRY: ghcr.io
+  REPOSITORY_NAME: ${{ github.repository }}
+
+jobs:
+  # define job to build and publish docker image
+  build-and-push-docker-image:
+    name: Build Docker image and push to repositories
+    # run only when code is compiling and tests are passing
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix: # Define a matrice to build each images and tag it
+        include:
+          - context: ./listingapi
+            image: listingapi
+          - context: ./pricemap-python
+            image: pricemap-python
+          - context: ./pricemap-typescript
+            image: pricemap-typescript
+
+    # steps to perform in job
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      # Github doesn't allow uppercase on registry name
+      - name: set lower case repository name
+        id: lowercase
+        run: echo "repository_name_lower_case=${REPOSITORY_NAME,,}" >>$GITHUB_OUTPUT
+        env:
+          REPOSITORY_NAME: '${{ env.REPOSITORY_NAME }}'
+
+      # Use Emulator to build images on different architectures
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@master
+
+      # Use BuildX to build the docker images
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to Github Packages
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ${{ env.REGISTRY }}/${{ steps.lowercase.outputs.repository_name_lower_case }}/${{ matrix.image }}
```

Comment:
@Yoan-MA https://github.com/docker/build-push-action/issues/237#issuecomment-848673650

https://github.com/docker/metadata-action#image-name-and-tag-sanitization

> To ease the integration in your workflow, this action will automatically:
> 
> Lowercase the image name